dspace/config/crosswalks/oai/metadataFormats/oai_openaire.xsl

```diff
@@ -0,0 +1,1484 @@
+    <xsl:template name="isDOI">
+        <xsl:param name="field"/>
+        <xsl:choose>
+            <xsl:when test="$field[contains(text(),'doi.org') or starts-with(text(),'10.')]">
+                <xsl:value-of select="true()"/>
+            </xsl:when>
+            <xsl:otherwise>
+                <xsl:value-of select="false()"/>
+            </xsl:otherwise>
+        </xsl:choose>
+    </xsl:template>
+
+    <!-- it will verify if a given field is an ORCID -->
+    <xsl:template name="isORCID">
+        <xsl:param name="field"/>
+        <xsl:choose>
+            <xsl:when test="$field[contains(text(),'orcid.org')]">
+                <xsl:value-of select="true()"/>
+            </xsl:when>
+            <xsl:otherwise>
+                <xsl:value-of select="false()"/>
+            </xsl:otherwise>
+        </xsl:choose>
+    </xsl:template>
+
+    <!-- it will verify if a given field is an URL -->
+    <xsl:template name="isURL">
+        <xsl:param name="field"/>
+        <xsl:variable name="lc_field">
+            <xsl:call-template name="lowercase">
+                <xsl:with-param name="value" select="$field"/>
+            </xsl:call-template>
+        </xsl:variable>
+        <xsl:choose>
+            <xsl:when test="$lc_field[starts-with(text(),'http://') or starts-with(text(),'https://')]">
+                <xsl:value-of select="true()"/>
+            </xsl:when>
+            <xsl:otherwise>
+                <xsl:value-of select="false()"/>
+            </xsl:otherwise>
+        </xsl:choose>
+    </xsl:template>
+
+    <!-- will try to resolve the field type based on the value -->
+    <xsl:template name="resolveFieldType">
+        <xsl:param name="field"/>
+        <!-- regexp not supported on XSLTv1 -->
+        <xsl:variable name="isHandle">
+            <xsl:call-template name="isHandle">
+                <xsl:with-param name="field" select="$field"/>
+            </xsl:call-template>
+        </xsl:variable>
+        <xsl:variable name="isDOI">
+            <xsl:call-template name="isDOI">
+                <xsl:with-param name="field" select="$field"/>
+            </xsl:call-template>
+        </xsl:variable>
+        <xsl:variable name="isURL">
+            <xsl:call-template name="isURL">
+                <xsl:with-param name="field" select="$field"/>
+            </xsl:call-template>
+        </xsl:variable>
+        <xsl:choose>
+            <xsl:when test="$isHandle = 'true'">
+                <xsl:text>Handle</xsl:text>
+            </xsl:when>
+            <xsl:when test="$isDOI = 'true'">
+                <xsl:text>DOI</xsl:text>
+            </xsl:when>
+            <xsl:when test="$isURL = 'true' and $isHandle = 'false'">
+                <xsl:text>URL</xsl:text>
+            </xsl:when>
+            <xsl:otherwise>
+                <xsl:text>N/A</xsl:text>
+            </xsl:otherwise>
+        </xsl:choose>
+    </xsl:template>
+
+    <xsl:template name="resolveResourceTypeGeneral">
```

Comment:
Please describe this template with a comment. It's *very helpful* to help understand the goal of the template, as this is a large XSL.  I appreciate that you've added such a comment for most templates already.

Reply:
I added comments for each existing templates. Thank you @tdonohue for the review!